### PR TITLE
#4148 add values_at method for CSV::Row

### PR DESCRIPTION
**Pull request by need47**
This PR adds the following methods to CSV::Row in (src/csv.cr)

values_at(*columns : Int)
values_at(*headers: String)

to retrieve multiple values at given columns or corresponding to headers.

A spec file was also provided at: spec/std/csv/csv_values_at_spec.cr

It's my first PR ever. If I was doing something wrong, please kindly let me know

### CODE REVIEW

File: src/csv.cr [hunk 1]
```diff
@@ -381,6 +381,22 @@ class CSV
       h
     end
 
+    # Returns a tuple of this row's values at given indices
+    # A negative index counts from the end.
+    # Raises `IndexError` if any column doesn't exist
+    # The behavior of returning a tuple is similar to `Hash#values_at` 
+    def values_at(*columns : Int)
+      columns.map { |column| self.[](column) }
```
Comment (Sija) [inline src/csv.cr]: `self[column]`

File: src/csv.cr [hunk 2]
```diff
@@ -381,6 +381,22 @@ class CSV
...
+    end
+
+    # Returns a tuple of this row's values corresponding to the given *headers*
+    # Raises `KeyError` if any header doesn't exist.
+    # Raises `CSV::Error` if headers were not requested
+    # The behavior of returning a tuple is similar to `Hash#values_at` 
+    def values_at(*headers : String)
+      headers.map { |header| self.[](header) }
```
Comment (Sija) [inline src/csv.cr]: ditto

**Comment by need47** [comment]
@Sija I have updated the PR according to your comment. My local test was good, but I saw that Travis CI had failed. Not sure whether it's related to my PR. Let me know if there is anything I can do.

**Comment by bcardiff** [comment]
Hi @need47, the failures are because of formatting. 

If you run
```
$ crystal tool format spec/std/csv/csv_values_at_spec.cr
$ crystal tool format src/csv.cr
```

There should be changes to submit and will make CI pass. At least linux one. There are currently some issues regarding OSX CI.

**Comment by need47** [comment]
All checks passed except for OSX CI. Just FYI, I did compile LLVM (version 3.9.1) without the OSX platform.

**Comment by need47** [comment]
I am reading the spec/std/csv/csv_spec.cr and found an interesting use of CSV.

```
it "can do new with block" do
    CSV.new(%(one, two\n1, 2\n3, 4\n5), headers: true, strip: true) do |csv|
      csv["one"].should eq("1")
      csv["two"].should eq("2")
      break
    end
  end
```

a csv object can call [] directly as if it was a row. So shall I also add CSV#values_at() methods, in addition to CSV::Row#values_at() in this PR?

**Comment by bcardiff** [comment]
@need47 Sure! mimic how `row_internal` is used in the other `[]` methods.

**Comment by need47** [comment]
I have added CSV#values_at and CI passed.

**Comment by bcardiff** [comment]
@need47 Sorry, I've just noticed that the specs should be at `spec/std/csv/csv_spec.cr`, would you move them when you have time?

**Comment by need47** [comment]
specs moved and CI passed.

**Comment by bcardiff** [comment]
Merged! Thanks @need47 for your first contribution to crystal

**Pull request merged**